hibernate-core/src/main/java/org/hibernate/persister/collection/BasicCollectionPersister.java

```diff
@@ -645,7 +645,13 @@ private RestrictedTableMutation<JdbcMutationOperation> generateDeleteRowAst(Muta
 				deleteBuilder.addKeyRestrictionsLeniently( pluralAttribute.getIndexDescriptor() );
 			}
 			else {
-				deleteBuilder.addKeyRestrictions( pluralAttribute.getElementDescriptor() );
```

Comment:
`elementColumnIsInPrimaryKey` essentially is a check if `!isNullable()`. We worked on supporting nullable columns in the past, but I don't remember where that work went. Maybe you didn't create a PR for that back then?

Anyway, I think the correct solution for this problem would be to handle nullable columns properly by using the `distinct from` predicate like we did in our earlier work.

Reply:
in this case is not true that `elementColumnIsInPrimaryKey essentially is a check if !isNullable()` we have columns that are null but they are PK. I just reintroduced what we do in 5. For what I remember the `distinc from` approach was super complicated